### PR TITLE
ref(errors): migrate existing legacy errors to ts

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Currently, our errors are in `.js`, which impacts our ability to make guarantees about the shape of the errors. This is an issue because we want to display error messages on our frontend based on what we return from our backend. In order to do this, we convert to `.ts` and guarantee the shape by extending from `BaseIsomerError`, which has a guarantee of said shape.

## Solution
1. Migrate from `.js` -> `.ts`

### CODE REVIEW

File: src/errors/ForbiddenError.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Import base error
+import { BaseIsomerError } from "@errors/BaseError"
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (kishore03109) [inline src/errors/ForbiddenError.ts:4]: Hmm if we feel as a team this is not impt eslint rule, could we change this at the global settings at a separate pr?

File: src/services/identity/__tests__/AuthService.spec.ts [hunk 2]
```diff
@@ -15,11 +14,6 @@ const AuthService = new _AuthService({
 })
 
 describe("Auth Service", () => {
-  const mocksessionData = {
```
Comment (kishore03109) [inline src/services/identity/__tests__/AuthService.spec.ts:18]: hmm why are we removing this ah

File: src/services/identity/__tests__/AuthService.spec.ts [hunk 3]
```diff
@@ -57,11 +51,8 @@ describe("Auth Service", () => {
 
   it("should call axios successfully and bubble the error when the status is not 403 or 404", async () => {
     // Arrange
-    const expected = {
-      response: { status: "400" },
-    }
     mockGitHubService.checkHasAccess.mockRejectedValueOnce(
-      new BadRequestError(expected)
+      new BadRequestError("")
```
Comment (kishore03109) [inline src/services/identity/__tests__/AuthService.spec.ts:55]: could I get context on why this returns a different shape now?

File: src/utils/json.ts [hunk 4]
```diff
@@ -4,5 +4,6 @@ import { UnprocessableError } from "@root/errors/UnprocessableError"
 
 export const safeJsonParse = Result.fromThrowable(
   JSON.parse,
-  () => new UnprocessableError()
+  () =>
+    new UnprocessableError("The given string could not be parsed into json!")
```
Comment (kishore03109) [inline src/utils/json.ts:8]: Hmm may I get context here? why are we adding this param here ah

File: src/errors/ForbiddenError.ts (hunk 1 above)
Comment (seaerchin) [reply src/errors/ForbiddenError.ts:4]: shure

File: src/services/identity/__tests__/AuthService.spec.ts (hunk 2 above)
Comment (seaerchin) [reply src/services/identity/__tests__/AuthService.spec.ts:18]: not used

File: src/services/identity/__tests__/AuthService.spec.ts (hunk 3 above)
Comment (seaerchin) [reply src/services/identity/__tests__/AuthService.spec.ts:55]: it still accepts the same shape - last time no type safety cos in `js` so it accepts object, but `message` is a `string`

File: src/utils/json.ts (hunk 4 above)
Comment (seaerchin) [reply src/utils/json.ts:8]: `message` is a required `string`

**Review by harishv7 — APPROVED** [review]

**Review by kishore03109 — APPROVED** [review]

**Pull request merged**